File: word/src/main/java/com/dreampany/word/ui/activity/NavigationActivity.java
package com.dreampany.word.ui.activity;

import android.os.Bundle;

import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelProviders;

import com.dreampany.frame.data.model.Task;
import com.dreampany.frame.misc.SmartAd;
import com.dreampany.frame.ui.activity.BaseBottomNavigationActivity;
import com.dreampany.frame.ui.callback.SearchViewCallback;
import com.dreampany.word.R;
import com.dreampany.word.databinding.ActivityNavigationBinding;
import com.dreampany.word.misc.Constants;
import com.dreampany.word.ui.fragment.FavoritesFragment;
import com.dreampany.word.ui.fragment.HomeFragment;
import com.dreampany.word.ui.fragment.MoreFragment;
import com.dreampany.word.ui.model.UiTask;
import com.dreampany.word.vm.LoaderViewModel;
import com.miguelcatalan.materialsearchview.MaterialSearchView;

import org.jetbrains.annotations.NotNull;

import javax.inject.Inject;

import dagger.Lazy;

/**
 * Created by Hawladar Roman on 5/24/2018.
 * BJIT Group
 * hawladar.roman@bjitgroup.com
 */
public class NavigationActivity extends BaseBottomNavigationActivity implements SearchViewCallback {

    @Inject
    Lazy<MoreFragment> moreFragment;
    @Inject
    Lazy<HomeFragment> homeFragment;
    @Inject
    Lazy<FavoritesFragment> favoritesFragment;
    @Inject
    ViewModelProvider.Factory factory;
    @Inject
    SmartAd ad;

    ActivityNavigationBinding bind;
    LoaderViewModel vm;

    @Override
    public int getLayoutId() {
        return R.layout.activity_navigation;
    }

    @Override
    public int getToolbarId() {
        return R.id.toolbar;
    }

    @Override
    public int getNavigationViewId() {
        return R.id.navigation_view;
    }

    @Override
    public int getDefaultSelectedNavigationItemId() {
        return R.id.item_home;
    }

    @Override
    public boolean isHomeUp() {
        return false;
    }

    @Override
    public boolean hasDoubleBackPressed() {
        return true;
    }

    @NotNull
    @Override
    public String getScreen() {
        return Constants.Screen.navigation(getApplicationContext());
    }

    @Override
    protected void onStartUi(Bundle state) {
        initView();
        ad.loadBanner(getClass().getSimpleName());
    }

    @Override
    protected void onStopUi() {
        ad.destroyBanner(getClass().getSimpleName());
    }

    @Override
    protected void onResume() {
        super.onResume();
        ad.resumeBanner(getClass().getSimpleName());
        vm.loads();
    }

    @Override
    protected void onPause() {
        ad.pauseBanner(getClass().getSimpleName());
        super.onPause();
    }

    @Override
    protected void onNavigationItem(int navigationItemId) {
        switch (navigationItemId) {
            case R.id.item_home:
                commitFragment(HomeFragment.class, homeFragment, R.id.layout);
                break;
            case R.id.item_favorites:
                commitFragment(FavoritesFragment.class, favoritesFragment, R.id.layout);
                break;
            case R.id.item_more:
                commitFragment(MoreFragment.class, moreFragment, R.id.layout);
                break;
        }
    }

    @Override
    public MaterialSearchView getSearchView() {
        return bind.searchView;
    }

    @Override
    public void execute(@NotNull Task<?> t) {

    }

    private void initView() {
        UiTask<?> uiTask = getCurrentTask(false);
        if (uiTask != null && uiTask.getType() != null && uiTask.getSubtype() != null) {
            openActivity(ToolsActivity.class, uiTask);
            return;
        }

        bind = (ActivityNavigationBinding) super.binding;
        ad.initAd(this,
                getClass().getSimpleName(),
                findViewById(R.id.adview),
                R.string.interstitial_ad_unit_id,
                R.string.rewarded_ad_unit_id);

        vm = ViewModelProviders.of(this, factory).get(LoaderViewModel.class);
    }
}

File: demo/src/main/kotlin/com/dreampany/demo/ui/activity/NavigationActivity.kt
package com.dreampany.demo.ui.activity

import android.os.Bundle
import com.dreampany.demo.R
import com.dreampany.demo.databinding.ActivityNavigationBinding
import com.dreampany.demo.misc.Constants
import com.dreampany.demo.ui.fragment.HomeFragment
import com.dreampany.demo.ui.fragment.MoreFragment
import com.dreampany.demo.ui.model.UiTask
import com.dreampany.frame.misc.SmartAd
import com.dreampany.frame.ui.activity.BaseBottomNavigationActivity
import dagger.Lazy
import javax.inject.Inject

/**
 * Created by Roman-372 on 7/24/2019
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
class NavigationActivity : BaseBottomNavigationActivity() {

    @Inject
    internal lateinit var homeFragment: Lazy<HomeFragment>
    @Inject
    internal lateinit var moreFragment: Lazy<MoreFragment>
    @Inject
    internal lateinit var ad: SmartAd

    private lateinit var bind: ActivityNavigationBinding

    override fun getLayoutId(): Int {
        return R.layout.activity_navigation
    }

    override fun getToolbarId(): Int {
        return R.id.toolbar
    }

    override fun getNavigationViewId(): Int {
        return R.id.navigation_view
    }

    override fun getDefaultSelectedNavigationItemId(): Int {
        return R.id.item_home
    }

    override fun isHomeUp(): Boolean {
        return false
    }

    override fun hasDoubleBackPressed(): Boolean {
        return true
    }

    override fun getScreen(): String {
        return Constants.navigation(applicationContext)
    }

    override fun onStartUi(state: Bundle?) {
        initView()
        ad.loadBanner(getScreen())
    }

    override fun onStopUi() {
        ad.destroyBanner(getScreen())
    }

    override fun onResume() {
        super.onResume()
        ad.resumeBanner(getScreen())
    }

    override fun onPause() {
        ad.pauseBanner(getScreen())
        super.onPause()
    }

    override fun onNavigationItem(navigationItemId: Int) {
        when (navigationItemId) {
            R.id.item_home -> commitFragment(HomeFragment::class.java, homeFragment, R.id.layout)
            R.id.item_more -> commitFragment(MoreFragment::class.java, moreFragment, R.id.layout)
        }
    }

/*    override fun getSearchView(): MaterialSearchView {
        return n
    }*/

    private fun initView() {
        bind = super.binding as ActivityNavigationBinding
        val uiTask = getCurrentTask<UiTask<*>>(false)
        if (uiTask != null && uiTask!!.type != null && uiTask!!.subtype != null) {
            openActivity(ToolsActivity::class.java, uiTask!!)
            return
        }
        ad.initAd(
            this,
            getScreen(),
            findViewById(R.id.adview),
            R.string.interstitial_ad_unit_id,
            R.string.rewarded_ad_unit_id
        )
    }
}